src/cljdoc/render.clj

```diff
@@ -40,7 +48,17 @@
                    first)
         [doc-type contents] (doctree/entry->type-and-content doc-p)
         top-bar-component (layout/top-bar version-entity (-> cache-bundle :version :scm :url))
-        sidebar-contents (sidebar/sidebar-contents route-params cache-bundle last-build)]
+        sidebar-contents (sidebar/sidebar-contents route-params cache-bundle last-build)
+        flatten-article-tree (->> doc-tree
+                                  (remove #(contains? #{"Readme" "Changelog"} (:title %)))
+                                  doctree/flatten*)
+        article-idx (->> flatten-article-tree
```

Comment:
Rather than working with indexes, you might be able to work directly with the article tree.

```clj
(partition 3 1 (concat [nil] (range 10) [nil]))
; => ((nil 0 1) (0 1 2) (1 2 3) (2 3 4) (3 4 5) (4 5 6) (5 6 7) (6 7 8) (7 8 9) (8 9 nil))
```

Replace `(range 10)` with `flatten-article-tree`. Then look for the seq where the middle element is the slug, then your `prev-page` and `next-page` are the first and the third element.

Reply:
wow, that's good, thank you!